lib/p2p/Pool.ts

```diff
@@ -712,18 +721,14 @@ class Pool extends EventEmitter {
         reject(err);
       };
 
-      this.server!.listen(this.listenPort || 0, '0.0.0.0').on('listening', () => {
-        const { address, port } = this.server!.address();
-        this.logger.info(`p2p server listening on ${address}:${port}`);
-
-        if (this.listenPort === 0) {
-          // we didn't specify a port and grabbed any available port
-          this.listenPort = port;
-        }
-
-        this.server!.removeListener('error', listenErrHandler);
-        resolve();
-      }).on('error', listenErrHandler);
+      this.interfaces!.forEach((uri) => {
+        this.server!.listen(uri.port, uri.host).on('listening', () => {
+          const { address, port } = this.server!.address();
+          this.logger.info(`p2p server listening on ${address}:${port}`);
+          resolve();
```

Comment:
This will resolve the promise after the first interface is being listened to, ideally we should wait for all of them to be listening before considering the method "complete." I think you should keep track of a counter like `listeningCount` which starts at 0, increments every time an interface starts listening, and then each time you increment it check it against `this.interfaces.length` and only if they are equal call `resolve()`.